e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -393,6 +418,51 @@ $systemIdentityPrincipal = az resource list -n $iotHubName --query [0].identity.
+# Uploading ROOT CA certificate to IoTHub and verifying
+##################################################################################################################################
+
+$certExits = az iot hub certificate list -g $ResourceGroup --hub-name $iotHubName --query "value[?name=='$hubUploadCertificateName']" --output tsv
+if ($certExits)
+{
+    Write-Host "`nDeleting existing certificate from IotHub"
+    $etag = az iot hub certificate show -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName --query 'etag'
+    az iot hub certificate delete -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName --etag $etag
+}
+Write-Host "`nUploading new certificate to IotHub"
+az iot hub certificate create -g $ResourceGroup --path $rootCertPath --hub-name $iotHubName --name $hubUploadCertificateName --output none
+
+$isVerified = az iot hub certificate show -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName --query 'properties.isVerified' --output tsv
+if ($isVerified -eq 'false')
+{
+    Write-Host "`nVerifying certificate uploaded to IotHub"
+    $etag = az iot hub certificate show -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName --query 'etag'
+    $requestedCommonName = az iot hub certificate generate-verification-code -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName -e $etag --query 'properties.verificationCode'
+    $verificationCertArgs = @{
+        "-DnsName"                       = $requestedCommonName;
+        "-CertStoreLocation"             = "cert:\LocalMachine\My";
+        "-NotAfter"                      = (get-date).AddYears(2);
+        "-TextExtension"                 = @("2.5.29.37={text}1.3.6.1.5.5.7.3.2,1.3.6.1.5.5.7.3.1", "2.5.29.19={text}ca=FALSE&pathlength=0"); 
+        "-Signer"                        = $rootCACert;
+    }
+    $verificationCert = New-SelfSignedCertificate @verificationCertArgs
+    Export-Certificate -cert $verificationCert -filePath $verificationCertPath -Type Cert | Out-Null
+    $etag = az iot hub certificate show -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName --query 'etag'
+    az iot hub certificate verify -g $ResourceGroup --hub-name $iotHubName --name $hubUploadCertificateName -e $etag --path $verificationCertPath --output none
+}
+
+##################################################################################################################################
+# Create device in IoTHub that uses a certificate signed by intermediate certificate
+##################################################################################################################################
+
+$iotHubCertChainDevice = az iot hub device-identity list -g vinageshe2etests --hub-name vinageshe2etests-hub --query "[?deviceId=='$iotHubCertChainDeviceCommonName'].deviceId" --output tsv 
```

Comment:
I think this is leftover test code?

Reply:
I did not add this in test as we run the tests in parallel and the device certificate is for a specific device only. Our tests can become flake if all tests try to create the same device.